src/k2/cmd/ycsb/ycsb_client.cpp

```diff
@@ -139,12 +139,18 @@ class Client {
         _global_id = (seastar::smp::count * _instance_id()) + seastar::this_shard_id();
 
         _num_keys = _num_records() + _num_inserts(); // key space is expanded to include expected inserts
+
         YCSBData::generateSchema(_num_fields());
         setupSchemaPointers();
 
         registerMetrics();
 
-        _benchFuture = _client.start().then([this] () { return _benchmark(); })
+        // This initial sleep is required because without it the reactor stall cased by the key
```

Comment:
Could we fix the keygen instead?

Reply:
It's not easy because I suspect most of the time is spent on the sort call that sorts all of the keys. We would need to think about if there is a better way to get the range ends for the collection.

Reply:
I tested it and the sorting does change the order of the keys. I made an issue so that we can look into it later: https://github.com/futurewei-cloud/chogori-platform/issues/262